### PR TITLE
adding app files: attempt 1

### PR DESCRIPTION
**Pull request by uurunov**
@matledzewicz

### CODE REVIEW

**Comment by uurunov** [comment]
Hi. I have finished all tasks except '(Extra) Category Sort Dropdown', '(Extra) Edit Task Form: Add Image support', and '(Extra) Create Task Form: Add task image'.

File: src/app/components/add-edit-category/add_edit_category.component.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    constructor(
+        private categoryService: CategoryService,
+        private router: Router,
+        private route: ActivatedRoute
+    ) {}
+
+    ngOnInit(): void {
+        this.route.params.subscribe( params => {
+            if (params.hasOwnProperty('id') && params['id'] > 0) {
```
Comment (gradzio) [inline src/app/components/add-edit-category/add_edit_category.component.ts:23]: imperative programming

File: src/app/components/add-edit-category/add_edit_category.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { CategoryService } from 'src/app/services/category.service';
+
+@Component({
+    selector: 'app-add-edit-category',
+    templateUrl: './add_edit_category.component.html',
+    styleUrls: ['./add_edit_category.component.scss']
+})
+
+export class AddEditCategoryComponent implements OnInit, OnDestroy {
+    name: string = '';
```
Comment (gradzio) [inline src/app/components/add-edit-category/add_edit_category.component.ts:12]: mutable component properties

File: src/app/models/task.model.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export interface CategoryTask {
+    name: string,
+    categoryId: number,
+    id: number,
+    imageUrl: string | any,
```
Comment (gradzio) [inline src/app/models/task.model.ts:5]: optional model properties

**Review by gradzio — CHANGES_REQUESTED** [review]

**Comment by uurunov** [comment]
@gradzio I have read your comments but I am not clear about the next thing I should do now to meet task requirement. I mean, I do not understand what is wrong with the code where you left comments, sir. Please, help and guide me.

**Review by uurunov — COMMENTED** [review]
@gradzio I have read your comments but I am not clear about the next thing I should do now to meet task requirement. I mean, I do not understand what is wrong with the code where you left comments, sir. Please, help and guide me.

**Comment by uurunov** [comment]
@gradzio I have read your comments but I am not clear about the next thing
I should do now to meet task requirement. I mean, I do not understand what
is wrong with the code where you left comments, sir. Please, help and guide
me.

**Pull request closed without merging**